langstream-cli/src/main/java/ai/langstream/cli/commands/docker/LocalRunApplicationCmd.java

```diff
@@ -0,0 +1,151 @@
+            description = "Start the S3 service")
+    private boolean startS3 = true;
+
+    @CommandLine.Option(
+            names = {"-s", "--secrets"},
+            description = "Secrets file path")
+    private String secretFilePath;
+
+    String appPath() {
+        return appPath;
+    }
+
+    String instanceFilePath() {
+        return instanceFilePath;
+    }
+
+    String secretFilePath() {
+        return secretFilePath;
+    }
+
+    @Override
+    @SneakyThrows
+    public void run() {
+        final File appDirectory = checkFileExistsOrDownload(appPath());
+        final File instanceFile = checkFileExistsOrDownload(instanceFilePath());
+        final File secretsFile = checkFileExistsOrDownload(secretFilePath());
+
+        log("Application directory: " + appDirectory.getAbsolutePath());
+        log("Instance file: " + instanceFile.getAbsolutePath());
+        log("Secrets file: " + secretsFile.getAbsolutePath());
+        log("Start broker: " + startBroker);
+        log("Start S3: " + startS3);
+
+        if ((appDirectory == null || instanceFile == null)) {
+            throw new IllegalArgumentException("application and instance files are required");
+        }
+
+        downloadDependencies(appDirectory.toPath());
+
+        String secretsContents;
+        String instanceContents;
+
+        try {
+
+            instanceContents =
+                    LocalFileReferenceResolver.resolveFileReferencesInYAMLFile(
+                            instanceFile.toPath());
+        } catch (Exception e) {
+            log(
+                    "Failed to resolve instance file references. Please double check the file path: "
+                            + instanceFile.toPath());
+            e.printStackTrace();
+            throw e;
+        }
+
+        try {
+
+            secretsContents =
+                    LocalFileReferenceResolver.resolveFileReferencesInYAMLFile(
+                            secretsFile.toPath());
+        } catch (Exception e) {
+            log(
+                    "Failed to resolve secrets file references. Please double check the file path: "
+                            + secretsFile.toPath());
+            throw e;
+        }
+
+        executeOnDocker(appDirectory, instanceContents, secretsContents, startBroker, startS3);
+    }
+
+    private void executeOnDocker(
+            File appDirectory,
+            String instanceContents,
+            String secretsContents,
+            boolean startBroker,
+            boolean startS3)
+            throws Exception {
+        File tmpInstanceFile = Files.createTempFile("instance", ".yaml").toFile();
+        Files.write(tmpInstanceFile.toPath(), instanceContents.getBytes(StandardCharsets.UTF_8));
+        File tmpSecretsFile = Files.createTempFile("secrets", ".yaml").toFile();
+        Files.write(tmpSecretsFile.toPath(), secretsContents.getBytes(StandardCharsets.UTF_8));
+        String imageName = "langstream/langstream-runtime-tester:latest-dev";
```

Comment:
we should make this configurable

Reply:
this should be configurable and it should use "some release", ideally the same version of the CLI